store/src/main/scala/geotrellis/store/GeoTrellisRasterSource.scala

```diff
@@ -174,25 +174,43 @@ object GeoTrellisRasterSource {
   }
 
   def readTiles(reader: CollectionLayerReader[LayerId], layerId: LayerId, extent: Extent, bands: Seq[Int]): Seq[(SpatialKey, MultibandTile)] with Metadata[TileLayerMetadata[SpatialKey]] = {
+    def spatialTileRead =
+      reader.query[SpatialKey, Tile, TileLayerMetadata[SpatialKey]](layerId)
+        .where(Intersects(extent))
+        .result
+        .withContext(tiles =>
+          // Convert single band tiles to multiband
+          tiles.map { case (key, tile) => (key, MultibandTile(tile)) }
+        )
+
+    def spatialMultibandTileRead =
+      reader.query[SpatialKey, MultibandTile, TileLayerMetadata[SpatialKey]](layerId)
+        .where(Intersects(extent))
+        .result
+        .withContext(tiles =>
+          tiles.map { case (key, tile) => (key, tile.subsetBands(bands)) }
+        )
+
     val header = reader.attributeStore.readHeader[LayerHeader](layerId)
-    (Class.forName(header.keyClass), Class.forName(header.valueClass)) match {
-      case (SpatialKeyClass, TileClass) =>
-        reader.query[SpatialKey, Tile, TileLayerMetadata[SpatialKey]](layerId)
-          .where(Intersects(extent))
-          .result
-          .withContext(tiles =>
-            // Convert single band tiles to multiband
-            tiles.map{ case(key, tile) => (key, MultibandTile(tile)) }
-          )
-      case (SpatialKeyClass, MultibandTileClass) =>
-        reader.query[SpatialKey, MultibandTile, TileLayerMetadata[SpatialKey]](layerId)
-          .where(Intersects(extent))
-          .result
-          .withContext(tiles =>
-            tiles.map{ case(key, tile) => (key, tile.subsetBands(bands)) }
-          )
-      case _ =>
-        throw new Exception(s"Unable to read single or multiband tiles from file: ${(header.keyClass, header.valueClass)}")
+
+    if (!header.keyClass.contains("spark")) {
+      (Class.forName(header.keyClass), Class.forName(header.valueClass)) match {
+        case (SpatialKeyClass, TileClass) => spatialTileRead
+        case (SpatialKeyClass, MultibandTileClass) => spatialMultibandTileRead
```

Comment:
This is mostly style comment that was generated from this line of thinking:

- We get the class class when the class loader is good and match
- If the class is old, we can't get a class class, so we have to check the strings
- And to do that, we have to do a sub-string check
- So wait, we made a class and did a sub-string check to avoid doing a string match?
- Is it slower, I don't know, probably doesn't matter...

TLDR: I think the original formulation is much clearer and potentially saves somebody thinking useless thoughts:

https://github.com/geotrellis/geotrellis-server/blob/89f297c54c12825512716bbcb7c65334a51d4014/core/src/main/scala/geotrellis/store/GeoTrellisRasterSourceLegacy.scala#L237-L244

Disregard this comment if you have a clear preference for doing it this way.